web/util.py

```diff
@@ -24,3 +25,19 @@ def flash_redirect(path: str, message: str | None = None, category="info"):
         flash(message, category)
 
     return redirect(path)
+
+
+def upload_file_to_printer(app, file, web_upload=False):
+    user_name = request.headers.get("User-Agent", "ankerctl").split(url_for('app_root'))[0]
+
+    with app.svc.borrow("filetransfer") as ft:
+        try:
+            ft.send_file(file, user_name)
+        except ConnectionError as err:
+            log.error(f"Connection error: {err}")
+            raise ConnectionError(err)
```

Comment:
This pattern is not good. In fact there are 3 problems :sweat_smile:

1) We are creating exception objects where the argument to the exception, is an exception of the same type.
2) Re-raising exceptions in python has an explicit syntax: `raise` (just the keyword, used inside an `except` clause). Otherwise, the backtrace into the actual error site is lost, and the exceptions seem to originate "here". (around python 3.7, I think, the language gained some support for working around this, but it's not kosher)
3) We are only catching these exceptions to log them, and then we have to catch them again when we call `upload_file_to_printer()`, so this doesn't much help us.

Reply:
✅